code/modules/mob/mob_helpers.dm

```diff
@@ -22,6 +22,27 @@
 			return 0
 	return 1
 
+/mob/proc/get_screen_colour()
+
+/mob/dead/observer/get_screen_colour()
+	return list()
+
+/mob/proc/update_client_colour(var/time = 10) //Update the mob's client.color if the colour matrix in the eyes (or whatever) is different.
+	if(!client)
+		return
+	client.colour_transition(get_screen_colour(), time = time)
+
+/mob/living/carbon/human/get_screen_colour() //Fetch the colour matrix from wherever (e.g. eyes) so it can be compared to client.color.
+	. = ..()
+	if(.)
+		return .
+	var/obj/item/clothing/glasses/worn_glasses = glasses
+	var/obj/item/organ/internal/eyes/eyes = get_int_organ(/obj/item/organ/internal/eyes)
+	if(istype(worn_glasses) && worn_glasses.color_view) //Check to see if they got those magic glasses and they're augmenting the colour of what the wearer sees. If they're not, color_view should be null.
+		return worn_glasses.color_view
+	else if(eyes && eyes.colourmatrix && !(eyes.robotic)) //The robotic eyes check is a redundant backup to the setting of colourmatrix during robotize() in code\modules\surgery\organs\organ_internal.dm.

```

Comment:
Yep, after all the structure changes and revisions as per your recommendations the `robotic` sanity check being removed has no ill effect so long as I move the robotic check [here](https://github.com/ParadiseSS13/Paradise/pull/6123/files#diff-c263363df08edeb8a6e85abb9a411321R151) up into [this line](https://github.com/ParadiseSS13/Paradise/pull/6123/files#diff-c263363df08edeb8a6e85abb9a411321R149) and adjust the indentation accordingly